packages/apex-parser-ast/src/utils/CaseInsensitiveMap.ts

```diff
@@ -33,6 +39,21 @@ export class CaseInsensitiveHashMap<V = any, R = [string, V]> extends HashMap<
     };
 
     super(entryOrRawElements, caseInsensitiveOptions);
+    // Populate originalKeys from entries if they were passed to constructor
+    // This ensures first-touch principle is maintained even when constructing with entries
+    if (entryOrRawElements) {
+      for (const entry of entryOrRawElements) {
+        const entryArray = Array.isArray(entry)
+          ? (entry as [string, V])
+          : (entry as unknown as [string, V]);
```

Comment:
While this works, it feels weird to read. Again, not a blocker but like, do we really want to have entry as unknown as [string, V] when we're trying to do that above anyways? is it ever possible it's anything else ?